### ISSUE TITLE
"Error: Unable to create dart snapshot for flutter tool" while trying to upgrade to flutter 2

### ISSUE THREAD

**Issue by Yosiwealth** [flutter/flutter#77413]
I tried upgrading flutter to the latest version using "flutter upgrade", I keep getting the error below.  Any help?

```
Building flutter tool...
Running pub upgrade...
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/platform-3.0.0/lib/src/interface/local_platform.dart: Error: The control character U+0000 can only be used in strings and comments.
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/platform-3.0.0/lib/src/interface/platform.dart: Error: The control character U+0000 can only be used in strings and comments.
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/platform-3.0.0/lib/src/testing/fake_platform.dart: Error: The control character U+0000 can only be used in strings and comments.
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:22:6: Error: Type 'Platform' not found.
    {Platform platform = const LocalPlatform()}) {
     ^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:54:3: Error: Type 'Platform' not found.
  Platform platform = const LocalPlatform(),
  ^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:22:6: Error: 'Platform' isn't a type.
    {Platform platform = const LocalPlatform()}) {
     ^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:22:32: Error: Method not found: 'LocalPlatform'.
    {Platform platform = const LocalPlatform()}) {
                               ^^^^^^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:54:3: Error: 'Platform' isn't a type.
  Platform platform = const LocalPlatform(),
  ^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:54:29: Error: Method not found: 'LocalPlatform'.
  Platform platform = const LocalPlatform(),
                            ^^^^^^^^^^^^^
Error: Unable to create dart snapshot for flutter tool.
Building flutter tool...
Running pub upgrade...
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/platform-3.0.0/lib/src/interface/local_platform.dart: Error: The control character U+0000 can only be used in strings and comments.
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/platform-3.0.0/lib/src/interface/platform.dart: Error: The control character U+0000 can only be used in strings and comments.
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/platform-3.0.0/lib/src/testing/fake_platform.dart: Error: The control character U+0000 can only be used in strings and comments.
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:22:6: Error: Type 'Platform' not found.
    {Platform platform = const LocalPlatform()}) {
     ^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:54:3: Error: Type 'Platform' not found.
  Platform platform = const LocalPlatform(),
  ^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:22:6: Error: 'Platform' isn't a type.
    {Platform platform = const LocalPlatform()}) {
     ^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:22:32: Error: Method not found: 'LocalPlatform'.
    {Platform platform = const LocalPlatform()}) {
                               ^^^^^^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:54:3: Error: 'Platform' isn't a type.
  Platform platform = const LocalPlatform(),
  ^^^^^^^^
AppData/Local/Pub/Cache/hosted/pub.dartlang.org/process-4.1.0/lib/src/interface/common.dart:54:29: Error: Method not found: 'LocalPlatform'.
  Platform platform = const LocalPlatform(),
                            ^^^^^^^^^^^^^
Error: Unable to create dart snapshot for flutter tool.
```

**Comment by darshankawar** [comment]
@Yosiwealth 
Can you try `flutter pub cache repair` or `flutter clean` and then `flutter upgrade` to see if it works ?

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by rw152** [comment]
I've been having this problem for a couple days now. I have tried:

- Using "git clone ...." to pull stable and install
- Using "git clone ...." to pull master and install
- Attempting to manually install by downloading the ```Master.zip``` from the repo, extracting, and manually installing
- Attempting to manually install by downloading ```Stable.zip``` from the repo, extracting, and manually installing

With no luck.

If I use git clone to pull stable, I can at least fire off the "flutter doctor" command.

If I download the zip file(s) and extract them to the flutter installation directory, running ```flutter [any command]``` causes the terminal to instantly crash.

I've attached the latest output from my ```flutter doctor``` attempt after using ```git clone https://github.com/flutter/flutter.git -b stable``` to try and get it running.

As a note, I haven't had any issues prior to installing Flutter 2 and upgrading to Null-Safety.
[broken.txt](https://github.com/flutter/flutter/files/6260211/broken.txt)
